src/sentry/spans/consumers/process/factory.py

```diff
@@ -113,14 +113,12 @@ def _build_snuba_span(relay_span: Mapping[str, Any]) -> MutableMapping[str, Any]
         sentry_tags["status"] = status
 
     if "status_code" in sentry_tags:
-        try:
-            sentry_tags["status_code"] = int(sentry_tags["status_code"])
-        except ValueError:
-            pass
+        sentry_tags["status_code"] = sentry_tags["status_code"]
 
-    snuba_span["sentry_tags"] = sentry_tags
+    snuba_span["sentry_tags"] = {k: str(v) for k, v in sentry_tags.items()}
```

Comment:
did we make sure we have no `None` otherwise those will be stuck in the pipeline? Maybe safer to do `{k: str(v)  for k, v in sentry_tags.items() if v is not None}`

Reply:
We don't have `None`. These would be logged to Sentry as validation errors b/c of https://github.com/getsentry/sentry-kafka-schemas/blob/main/schemas/snuba-spans.v1.schema.json#L141. Afaik none is not a type we support for tag value in any of our datasets.

On the off chance there is a none somehow, I think it's fine to cast to `"None"` anyway rather than drop it.

Reply:
I'm also checking https://github.com/getsentry/sentry/pull/56362/files#diff-973384a59a8a97829c8a4ceb385fc4fa74a3ac9b250191da9e53ef22f95a518eR106-R107 the values are not `None` when building the `dict`.